OutOfSchool/OutOfSchool.WebApi/Services/BlockedProviderParentService.cs

```diff
@@ -106,7 +106,10 @@ public async Task<Result<BlockedProviderParentDto>> Unblock(BlockedProviderParen
             throw new ArgumentNullException(nameof(blockedProviderParentUnblockDto));
         }
 
-        var currentBlock = await GetBlock(blockedProviderParentUnblockDto.ParentId, blockedProviderParentUnblockDto.ProviderId).ConfigureAwait(false);
+        var currentBlock = (await GetBlockedProviderParentEntities(
```

Comment:
(Perf issue) You perform reading db for the same data twice - once here (using as a key `ParentId`, and `ProviderId`) and other time in `blockedProviderParentRepository.UnBlock()`. It's bit inefficient. 

Here you get everything and then specify `FirstOrDefault()` for the `IEnumerable`, while you could use `FirstOrDefaultAsync()` for the `IQueryable`.

Similar is done below in the `GetBlock()` and `IsBlocked()` - instead of using `IQueryable.FirstOrDefaultAsync()`/`IEntityAddOnlyRepository.Any()` (both async) - both time `IEnumerable`s `FirstOrDefault()`/`Any()` are used (that are synchronous and thus you have to use `async`/`await` that affects performance even more).

Reply:
Fixed. Thank you!